test/base_test.py

```diff
@@ -25,6 +25,22 @@ def decode_response(response: BaseServer.Response, format: str = "json"):
     raise ValueError(f"Unsupported format: {format}")
 
 
+def decode_array_response(
+    response: BaseServer.Response,
+    format: str,
+    dtype: str,
+    shape: Tuple[int],
+) -> np.ndarray:
+    """Decode data array response content according to given information"""
+    content_type = [h[1] for h in response.headers if h[0] == "Content-Type"][0]
```

Comment:
Just something I thought of:
```suggestion
    content_type = {h[0]: h[1] for h in response.headers}["Content-Type"]
```